File: backend/semgrep_rules/tests/javascript/kisa_sw_27_improper_signature_verification.js
const jwt = require("jsonwebtoken");
function unsafe(token) {
  // ruleid: kisa.sw27.javascript.decode-without-signature-verification
  return jwt.decode(token);
}
function safe(token, key) {
  // ok: kisa.sw27.javascript.decode-without-signature-verification
  return jwt.verify(token, key, { algorithms: ["RS256"] });
}

File: backend/semgrep_rules/tests/python/kisa_sw_27_improper_signature_verification.py
import jwt
def unsafe(token):
    # ruleid: kisa.sw27.python.signature-verification-disabled
    return jwt.decode(token, options={"verify_signature": False})
def safe(token, key):
    # ok: kisa.sw27.python.signature-verification-disabled
    return jwt.decode(token, key, algorithms=["RS256"])
